apidef/oas/operation.go

```diff
@@ -0,0 +1,157 @@
+package oas
+
+import (
+	"strings"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+type Operations map[string]*Operation
+
+type Operation struct {
+	Allow                *Allowance `bson:"allow,omitempty" json:"allow,omitempty"`
+	Block                *Allowance `bson:"block,omitempty" json:"block,omitempty"`
+	IgnoreAuthentication *Allowance `bson:"ignoreAuthentication,omitempty" json:"ignoreAuthentication,omitempty"`
+}
+
+const (
+	allow                AllowanceType = 0
+	block                AllowanceType = 1
+	ignoreAuthentication AllowanceType = 2
+)
+
+type AllowanceType int
+
+func (s *OAS) fillPathsAndOperations(ep apidef.ExtendedPathsSet) {
+	if s.Paths == nil {
+		s.Paths = make(openapi3.Paths)
+	}
+
+	s.fillAllowance(ep.WhiteList, allow)
+	s.fillAllowance(ep.BlackList, block)
+	s.fillAllowance(ep.Ignored, ignoreAuthentication)
+
+	if len(s.Paths) == 0 {
+		s.Paths = nil
+	}
+}
+
+func (s *OAS) extractPathsAndOperations(ep *apidef.ExtendedPathsSet) {
+	tykOperations := s.getTykOperations()
+	if len(tykOperations) == 0 {
+		return
+	}
+
+	for id, tykOp := range tykOperations {
+		for path, pathItem := range s.Paths {
+			for method, operation := range pathItem.Operations() {
+				if id == operation.OperationID {
```

Comment:
Do we even need to loop over `tykOperations` or can we just do `tykOp, ok := tykOperations[operation.OperationID]` instead of `id == operation.OperationID`?

Reply:
The idea of putting `tykOperations` to top was because it is source of truth. Actually, I forgot to add break statements there to making this approach meaningful. I will append them so that way will make sense.

Reply:
The idea of putting `tykOperations` to top was because it is source of truth. Actually, I forgot to add break statement there to making this approach meaningful. I will append it so that way will make sense. Otherwise, `tykOperations[operation.OperationID]` causes unnecessary iterations.

Reply:
The idea of putting `tykOperations` to top was because it is source of truth. Actually, I forgot to add break statement there to making this approach meaningful. I will append it so that way will make sense. Otherwise, `tykOperations[operation.OperationID]` causes unnecessary iterations.

Reply:
If tykOperations is local to the function, we can also delete(tykOperations, operation.OperationID) - unless we particularly want to overwrite conflicts? Should we raise a warning in case of duplicated operationIDs? The code would be more efficient, if we optimize for uniqueness of operationIDs...